### PR TITLE
Feat/Android locale support

### PR DESCRIPTION
**Pull request by Inbal-Tish**
# Summary

Currently, DateTimePickers on Android use system locale and not the app one, so a user with English system language and Spanish app language would see a picker in English instead of Spanish. See issue #82 
I added support for passing 'locale' prop, the same way as in iOS, and get the picker in that locale (date picker will localize the dialog header and calendar's month and days of the week views).
As part of this effort, I added the 'positiveButtonLabel' and 'negativeButtonLabel' props to help pass localized strings to the buttons labels.

Uncomment App.js lines #214-216 to pass the appropriate props to see the example.

Screen Shot 2020-07-27 at 17 00 30

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| Android |    ✅|

## Checklist

- [✅ ] I have tested this on a device and a simulator
- [✅ ] I added the documentation in `README.md`
- [✅ ] I mentioned this change in `CHANGELOG.md`
- [✅ ] I updated the typed files (TS and Flow)
- [ ✅] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#Wed Jul 22 14:26:15 IDT 2020
```
Comment (vonovak) [inline android/gradle/wrapper/gradle-wrapper.properties:1]: why do you belive these gradle files should be included in the repo?

File: android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java [hunk 2]
```diff
@@ -86,6 +87,10 @@ DatePickerDialog getDialog(
                   display
           );
         default:
+          if (args != null && args.containsKey(RNConstants.ARG_LOCALE)) {
+            // for header (NOTE: will change the app locale)
+            Locale.setDefault(getLocale(args));
```
Comment (vonovak) [inline android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java:92]: It's quite possible there is no other way to localize the pickers, but this seems like an action that can have far-reaching consequences. 

on [ios](https://developer.apple.com/documentation/uikit/uidatepicker#2281467) the prop just "overrides the system default with a specific locale." - that is a "safe way" of specifying locale.

Here, the call would have side effects beyond the datepicker, and I'm not sure we want to take that path. Seem to me that code like this should probably be called from userland upon app start. What do you think?


(also applies to `activityContext.getResources().getConfiguration().setLocale(getLocale(args))`).

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (Inbal-Tish) [reply android/gradle/wrapper/gradle-wrapper.properties:1]: These are probably auto-generated when running the example project. Do you think they are not necessary?

File: android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java (hunk 2 above)
Comment (Inbal-Tish) [reply android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java:92]: I know and I do understand your point. I was looking for a way to localize it without localizing the entire app but couldn't find another way. I was also thinking that passing the 'locale' prop to localize the picker is made for an already localized app when the user what to have the picker in the app's locale, not the device's. Thus no real harm is done when we change the locale this way. We can add a note to these side-effects when using this feature. What do you think?

Comment (vonovak) [reply android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java:92]: I would prefer to not do it this way. I'm not even sure setting locale is a job for this module, maybe it should be done in the scope of some different package..

If we're to do this, I would prefer for this locale-changing logic to be exposed via a `@ReactMethod` that users can call. That will express the side effect more clearly.

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (vonovak) [reply android/gradle/wrapper/gradle-wrapper.properties:1]: I think they are not necessary

**Comment by vonovak** [comment]
@Inbal-Tish would you please address the review? I'd like to merge the PR once the comments are addressed. Thank you!

**Review by hondaholics — APPROVED** [review]

**Review by clementbmn — APPROVED** [review]

**Comment by Inbal-Tish** [comment]
@vonovak Hi. Sorry I didn't have the time to get to it again. Hopefully soon...

**Comment by marvin-42** [comment]
Any progress with this?

**Comment by pauloaapereira** [comment]
Any progress on this now in 2021? :(
Thanks

**Comment by nartenergy** [comment]
Any update on this??

**Comment by vonovak** [comment]
hello to everyone commenting; as I explained in https://github.com/react-native-datetimepicker/datetimepicker/issues/313, we're looking for backers that would support the development of this module and due to reasons explained in the post, I no longer actively develop this module. In order to move this forward, I invite you to do any of the following:

- take this PR, address comments from code review and open another PR that me or other maintainers will merge
- support the module via one of the offered sponsorship options and once we collect enough balance, I would take a look at it.
- (shamlesss plug) you can hire me. I'm sorry if this sounds like I'm trying to sell anything; in fact, I have plenty of work to do elsewhere, I am just listing the options :)

Thank you :)

**Comment by vonovak** [comment]
closed by https://github.com/react-native-datetimepicker/datetimepicker/pull/540 (v5 release)

**Comment by KMathisGit** [comment]
Should re-open this as the v5 release (removing locale prop) was reverted in v5.1.0.

**Pull request closed without merging**